components/GoogleMap.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <div id="Map">
+    <!-- The Google Maps API will make use of this div. -->
+  </div>
+</template>
+
+<script>
+// TODO: I don't know why scriptjs needs to be defined here
+let scriptjs
+
+export default {
+  data: function () {
+    return {
+      map: false
+    }
+  },
+  mounted() {
+    // TODO: A reference error will occur if scriptjs is define above and not here
+    scriptjs = require('scriptjs')
+    // Load the Google Maps API
+    scriptjs('https://maps.googleapis.com/maps/api/js?key=' + process.env.googleMapsApi, () => {
+      this.initMap()
+    })
+  },
+  methods: {
+    // The callback after the Google Maps JS API loads
+    initMap: () => {
```

Comment:
Declaring arrow functions in methods can cause issues, just use the short hand notation i.e.
`initMap() {`